pkg/registry/pod/etcd/etcd.go

```diff
@@ -64,23 +64,14 @@ func NewStorage(s storage.Interface, useCacher bool, k client.ConnectionInfoGett
 	prefix := "/pods"
 
 	storageInterface := s
+	newListFunc := func() runtime.Object { return &api.PodList{} }
 	if useCacher {
-		config := storage.CacherConfig{
-			CacheCapacity:  1000,
-			Storage:        s,
-			Type:           &api.Pod{},
-			ResourcePrefix: prefix,
-			KeyFunc: func(obj runtime.Object) (string, error) {
-				return storage.NamespaceKeyFunc(prefix, obj)
-			},
-			NewListFunc: func() runtime.Object { return &api.PodList{} },
-		}
-		storageInterface = storage.NewCacher(config)
+		storageInterface = storage.NewCacher(1000, s, nil, &api.Pod{}, prefix, true, newListFunc)
```

Comment:
(Discussed in person: I suggested passing a construction function in instead of the boolean)